### PR TITLE
Update bash-completion

### PR DESCRIPTION
**Pull request by JunweiSUN**
Update tools/bash-completion, adding new commands and options, resolve previous bugs.
Also update some lines in docs/en_US/Nnictl.md and python version requirements.

### CODE REVIEW

File: deployment/pypi/setup.py [hunk 1]
```diff
@@ -15,6 +17,34 @@
 else:
     raise NotImplementedError('current platform {} not supported'.format(os_type))
 
+class AutoCompletion(install):
+    def run(self):
+        COMP_URL = 'https://raw.githubusercontent.com/microsoft/nni/master/tools/bash-completion'
```
Comment (QuanluZhang) [inline deployment/pypi/setup.py:22]: it is not good to use master branch here

File: deployment/pypi/setup.py [hunk 2]
```diff
@@ -15,6 +17,34 @@
...
+        if os_type == 'Linux':
+            HOME = os.environ.get('HOME')
+            if not HOME:
+                install.run(self)
+                return # can not get $HOME, abort.
+            if not os.geteuid(): # run as root
+                BASH_COMP_PREFIX = '/usr/share/bash-completion/completions'
```
Comment (QuanluZhang) [inline deployment/pypi/setup.py:29]: have you tried macos?

File: docs/en_US/Tutorial/Nnictl.md [hunk 3]
```diff
@@ -704,6 +704,21 @@ Debug mode will disable version check function in Trialkeeper.
...
 
 * __nnictl webui url__
+  * Description
+
+    Show an experiment's webui url
+
+  * Usage
+
+    ```bash
+    nnictl webui url [options]
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/Nnictl.md:714]: `options` -> `experiment_id`

File: deployment/pypi/setup.py (hunk 1 above)
Comment (QuanluZhang) [reply deployment/pypi/setup.py:22]: why not pack it into python package?

File: tools/bash-completion [hunk 4]
```diff
@@ -1,38 +1,42 @@
 # list of commands/arguments
```
Comment (QuanluZhang) [inline tools/bash-completion:1]: is it possible to automatically generate this file? @liuzhe-lz

File: deployment/pypi/setup.py (hunk 1 above)
Comment (JunweiSUN) [reply deployment/pypi/setup.py:22]: Sure, will change to a specific release branch.
If we pack it into python package, it is hard to locate the root of the python package, so I choose to directly download it.

File: tools/bash-completion (hunk 4 above)
Comment (JunweiSUN) [reply tools/bash-completion:1]: Maybe we can just write the whole file's content to the correct location.

File: docs/en_US/Tutorial/Nnictl.md (hunk 3 above)
Comment (JunweiSUN) [reply docs/en_US/Tutorial/Nnictl.md:714]: I think here we should follow the same format as above.

File: deployment/pypi/setup.py (hunk 2 above)
Comment (QuanluZhang) [reply deployment/pypi/setup.py:29]: sure, it is ok to not support macOS, but what is the consequence of installing nni in macOS? this directory will be created?

File: docs/en_US/Tutorial/Nnictl.md (hunk 3 above)
Comment (QuanluZhang) [reply docs/en_US/Tutorial/Nnictl.md:714]: i agree, that's why i suggest changing `options` to `experiment_id`...

File: deployment/pypi/setup.py (hunk 2 above)
Comment (JunweiSUN) [reply deployment/pypi/setup.py:29]: Only created in Linux. Line 23 do os version check.

Comment (QuanluZhang) [reply deployment/pypi/setup.py:29]: got it, thanks

File: deployment/pypi/setup.py [hunk 5]
```diff
@@ -48,7 +79,7 @@
         'nnicli': '../../src/sdk/pycli/nnicli'
     },
     package_data = {'nni': ['**/requirements.txt']},
-    python_requires = '>=3.5',
+    python_requires = '>=3.6',
```
Comment (SparkSnail) [inline deployment/pypi/setup.py:82]: why upgrade python?

File: deployment/pypi/setup.py [hunk 6]
```diff
@@ -15,6 +17,35 @@
...
     raise NotImplementedError('current platform {} not supported'.format(os_type))
 
+class AutoCompletion(install):
+    def run(self):
+        RELEASE_VERSION = os.popen('git describe --tags --abbrev=0').read().strip()
+        COMP_URL = 'https://raw.githubusercontent.com/microsoft/nni/{}/tools/bash-completion'.format(RELEASE_VERSION)
+        if os_type == 'Linux':
+            HOME = os.environ.get('HOME')
+            if not HOME:
+                install.run(self)
```
Comment (SparkSnail) [inline deployment/pypi/setup.py:27]: what is `install.run(self)` used for here?

Comment (JunweiSUN) [reply deployment/pypi/setup.py:27]: Not quiet sure, seems that this line actually do the install action. Every example about how to use cmdclass has this line, like this [link](https://stackoverflow.com/questions/21915469/python-setuptools-install-requires-is-ignored-when-overriding-cmdclass)

File: deployment/pypi/setup.py (hunk 5 above)
Comment (JunweiSUN) [reply deployment/pypi/setup.py:82]: v1.8 no longer support python 3.5, see #2778

File: deployment/pypi/setup.py [hunk 7]
```diff
@@ -15,6 +17,35 @@
 else:
     raise NotImplementedError('current platform {} not supported'.format(os_type))
 
+class AutoCompletion(install):
```
Comment (scarlett2018) [inline deployment/pypi/setup.py:20]: better to remove installation, and update doc to explain how to enable bash completion, or output message after nni installation to tell users to enable bash completion

Comment (JunweiSUN) [reply deployment/pypi/setup.py]: Installation removed. An [tutorial documentation](https://github.com/microsoft/nni/blob/0153b3f6eff4cce807857bcee30c872fab63caf2/docs/en_US/CommunitySharings/AutoCompletion.md) has been added.

**Comment by JunweiSUN** [comment]
@QuanluZhang @SparkSnail @scarlett2018 @liuzhe-lz Since this feature will be refactored in v2.0, now I provide a tutorial to guide users how to install bash-completion instead of installing it with `pip`. Please review again when you have time.

**Review by liuzhe-lz — APPROVED** [review]

File: docs/en_US/CommunitySharings/AutoCompletion.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
+# Auto Completion
```
Comment (QuanluZhang) [inline docs/en_US/CommunitySharings/AutoCompletion.md:1]: "Auto Completion" -> "Auto Completion of nnictl Commands"

**Review by QuanluZhang — APPROVED** [review]

**Review by SparkSnail — APPROVED** [review]

File: docs/en_US/CommunitySharings/AutoCompletion.md (hunk 8 above)
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/AutoCompletion.md]: Updated.

File: docs/en_US/CommunitySharings/AutoCompletion.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
+# Auto Completion for nnictl Commands
```
Comment (QuanluZhang) [inline docs/en_US/CommunitySharings/AutoCompletion.md:1]: please mention it after installation doc to let users know there is auto completion support. for example, after QuickStart installation

Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/AutoCompletion.md:1]: Sure, has added.

**Pull request merged**